src/lib/storage/value_segment.cpp

```diff
@@ -119,7 +119,6 @@ template <typename T>
 void ValueSegment<T>::resize(const size_t size) {
   DebugAssert(size > _values.size() && size <= _values.capacity(),
```

Comment:
Shouldn't a `resize` to `size` be ok?
```suggestion
  DebugAssert(size >= _values.size() && size <= _values.capacity(),
```

Reply:
The message might be slightly misleading, but I don't see the need splitting up the two. While resizing it to the current size is not really wrong, doing so is a good sign that you are doing something wrong.

Reply:
But should you resize to capacity?